prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -713,4 +732,166 @@ class ActionHistory {
             }
         }
     }
+
+    /**
+     * Creates a string that will be used to populate the action_response column of an action
+     * after that action has been completed
+     * @param options Message options passed in
+     * @param warnings Store of warnings generated during the pipeline
+     * @param errors Store of errors generated during the pipeline
+     * @param verbose If true, all item routing details will be included in the response
+     * @param actionHistory The ongoing action history instance being used
+     * @param report The report this response body is for
+     */
+    fun createResponseBody(
+        options: Options,
+        warnings: List<ResultDetail>,
+        errors: List<ResultDetail>,
+        verbose: Boolean,
+        actionHistory: ActionHistory? = null,
+        report: Report? = null
+    ): String {
+        val factory = JsonFactory()
+        val outStream = ByteArrayOutputStream()
+        factory.createGenerator(outStream).use {
+            it.useDefaultPrettyPrinter()
+            it.writeStartObject()
+            if (report != null) {
+                it.writeStringField("id", report.id.toString())
+                it.writeStringField("timestamp", DateTimeFormatter.ISO_INSTANT.format(Instant.now()))
+                it.writeStringField("topic", report.schema.topic)
+                it.writeNumberField("reportItemCount", report.itemCount)
+                // TODO: need to get the correct path and set the ENDPOINT_BASE correctly, blocked by waiting for
+                //  the historyApi to be ready.
+                // it.writeStringField("location", "[base path]/${report.id}")
```

Comment:
Is this dead code or part of the comment above?  If dead code then remove

Reply:
Part of the comment above - once we know the endpoint for a file via the history API it will need to be uncommented and populated correctly.